data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/CountAggregateAction.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-Limense-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.aggregate.actions;
+
+import org.opensearch.dataprepper.model.metric.JacksonSum;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.plugins.otel.codec.OTelProtoCodec;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateAction;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionInput;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionResponse;
+import org.opensearch.dataprepper.plugins.processor.aggregate.GroupState;
+import io.opentelemetry.proto.metrics.v1.AggregationTemporality;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Optional;
+
+/**
+ * An AggregateAction that combines multiple Events into a single Event. This action will add the unique keys of each smaller Event to the overall groupState,
+ * and will create a combined Event from the groupState on concludeGroup. If smaller Events have the same keys, then these keys will be overwritten with the keys of the
+ * most recently handled Event.
+ * @since 1.3
+ */
+@DataPrepperPlugin(name = "count", pluginType = AggregateAction.class, pluginConfigurationType = CountAggregateActionConfig.class)
+public class CountAggregateAction implements AggregateAction {
+    static final String EVENT_TYPE = "event";
+    static final String SUM_METRIC_NAME = "count";
+    static final String SUM_METRIC_DESCRIPTION = "Number of events";
+    static final String SUM_METRIC_UNIT = "1";
+    static final boolean SUM_METRIC_IS_MONOTONIC = true;
+    public final String countKey;
+    public final String outputFormat;
+    private long startTimeNanos;
```

Comment:
This start time is not in the `groupState`. Thus it will be shared by each group. I think you want to add this into the `groupState`.

Reply:
I agree. I am suggesting that startTime and endTime for "raw" may be added at aggregate processor level controlled by some thing like "add_keys". That's what we have discussed previously